test/CMakeLists.txt

```diff
@@ -64,8 +64,8 @@ endforeach ()
 
 if (Python_NumPy_FOUND AND (SWIG_FOUND OR (EXISTS ${PROJECT_SOURCE_DIR}/src/swig/nlopt-python.cpp)))
   set (PYINSTALLCHECK_ENVIRONMENT "LD_LIBRARY_PATH=${PROJECT_BINARY_DIR}/src/swig"
-                                  "PYTHONPATH=${PROJECT_BINARY_DIR}/src/swig"
-    )
+                                  "PYTHONPATH=$<SHELL_PATH:${PROJECT_BINARY_DIR}/src/swig/python;${PROJECT_BINARY_DIR}/src/swig>"
+      )
 
```

Comment:
now the location for the manual dll copy must be changed too (probably not great):
```
diff --git a/.github/workflows/build.yml b/.github/workflows/build.yml
index 4abae4d..2ba7900 100644
--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -41,7 +41,7 @@ jobs:
           pip install numpy
           cmake -DCMAKE_INSTALL_PREFIX=%cd:\=/%/install -DNLOPT_TESTS=ON .
           cmake --build . --config Release --target install
-          xcopy %cd%\install\bin\nlopt.dll %cd%\src\swig
-          xcopy %cd%\install\Lib\site-packages\_nlopt*.pyd %cd%\src\swig
+          xcopy %cd%\install\bin\nlopt.dll %cd%\src\swig\python
+          xcopy %cd%\install\Lib\site-packages\_nlopt*.pyd %cd%\src\swig\python
           dir /p %cd%\src\swig
           ctest -C Release --output-on-failure --timeout 100
```
I guess the same thing is missing that makes the java tests fails too on windows

Reply:
Yes, the Java tests also fail on Windows for the same reason (a DLL that was not found). I will look into it. Unfortunately, in my experience, DLL search paths are a particular pain on Windows: It supports neither RPATH/RPATH nor LD_LIBRARY_PATH. Sometimes, it finds DLLs in PATH, except when it does not. And sometimes it honors search paths set somewhere in the registry, except when it does not. Copying the DLL multiple times is often the only reliable way. I will tweak the xcopy lines and hope I get them right.

Reply:
This should be fixed in the latest revision (commit 062dd630c17cd0ad48a90a889f8cf737c7fac2b7): The Python stuff gets copied to the new location. For Java, `java.library.path` seems to work to get Java to find `nloptjni.dll`, but then `nlopt.dll` (on which `nloptjni.dll` depends) was not found, so I now copy `nlopt.dll` to where `nloptjni.dll` is found.

I also fixed the installed location of `nloptjni.dll` (should be in `bin`, not `lib`, same as `nlopt.dll`), but the tests are not using it from there, but from the build tree.